File: MergeSortedArray.py
class Solution:
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: void Do not return anything, modify nums1 in-place instead.
        """
        r1 = m-1
        r2 = n-1
        r = m+n-1
        while r1>=0 and r2>=0:
            if nums1[r1]>nums2[r2]:
                nums1[r] = nums1[r1]
                r -= 1
                r1 -= 1
            else:
                nums1[r] = nums2[r2]
                r -= 1
                r2 -= 1
        while r2>=0:
            nums1[r] = nums2[r2]
            r -= 1
            r2 -= 1
            
if __name__=='__main__':
    nums1 = list(input())
    nums2 = list(input())
    m = len(nums1)
    n = len(nums2)
    nums1+=[0 for i in range(n)]
    solution = Solution()
    solution.merge(nums1, m, nums2, n)
    print(nums1)

File: MergeSortedArray.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        for(int i=m+n-1;i>=n;i--){
            nums1[i]=nums1[i-n];
        }
        
        int l1=n, l2=0, l=0;
        while(l1<m+n&&l2<n){
            if(nums1[l1]>nums2[l2]) nums1[l++]=nums2[l2++];
            else nums1[l++]=nums1[l1++];
        }
        while(l2<n){
            nums1[l++]=nums2[l2++];
        }
    }
};

int main(){
    int m, n;
    cin>>m>>n;
    
    vector<int> nums1(m+n, 0);
    vector<int> nums2;
    
    for(int i=0;i<m;i++){
        cin>>nums1[i];
    }
    for(int i=0;i<n;i++){
        int num;
        cin>>num;
        nums2.push_back(num);
    }
    
    Solution *solution=new Solution();
    solution->merge(nums1, m, nums2, n);
    
    for(int i=0;i<m+n;i++){
        cout<<nums1[i]<<" ";
    }
    
    return 0;
}
